src/engine.rs

```diff
@@ -256,17 +274,18 @@ impl Engine {
     }
 
     /// Removes an account.
-    pub fn remove_account(address: &Address) {
+    pub fn remove_account(address: &Address, generation: u32) {
         Self::remove_nonce(address);
         Self::remove_balance(address);
         Self::remove_code(address);
         Self::remove_all_storage(address);
+        Self::set_generation(address, generation + 1);
```

Comment:
Move this call inside `remove_all_storage`

Reply:
Good call, done.